otter/integration/lib/autoscale.py

```diff
@@ -72,6 +84,75 @@ def decide(resp):
             .addCallback(decide)
         )
 
+    def wait_for_N_servers(
+        self, rcs, servers_desired, period=10, timeout=600, clock=None
+    ):
+        """Waits for the desired number of servers.
+
+        :param TestResources rcs: The integration test resources instance.
+            This provides useful information to complete the request, like
+            which endpoint to use to make the API request.
+        :param int servers_desired: The number of servers to wait for.
+            It must be greater than or equal to one.
+        :param int period: The number of seconds between poll attempts.
+            If unspecified, defaults to 10 seconds.
+        :param int timeout: The number of seconds to wait before giving up.
+            Defaults to 600 seconds (10 minutes).
+        :param twisted.internet.interfaces.IReactorTime clock: If provided,
+            the clock to use for scheduling things.  Defaults to `reactor`
+            if not specified.
+
+        :return: If the operation succeeds, the same instance of TestResources.
+            Otherwise, an exception is raised.
+        """
+
+        clock = clock or reactor
+
+        class Looper(object):
```

Comment:
You could do 
```python
retry(check_servers, can_retry=transient_errors_except(UnknownScalingGroup), next_interval=repeating_interval(10))
```
where `UnknownScalingGroup` exception is raised on 404 instead of implementing your own `Looper`.